walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,95 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //input: events collection and request for meeting
+    //output: collection of all times the proposed meeting could be held at
+
+    Collection<String> attendeesList = request.getAttendees();
+    long meetingDuration = request.getDuration();
+    List<TimeRange> proposedTimeRanges = new ArrayList<TimeRange>();
+    List<TimeRange> clashAttendeeEvents = new ArrayList<TimeRange>();
+
+    int startDay = TimeRange.START_OF_DAY;
+    int endDay = TimeRange.END_OF_DAY;
+
+    /*loop through all the planned events and find the relevant ones 
+    (where there's an attendee clash)
+    */
+    for (Event event: events) {
+    //get the attendees from the specific event you're checking the attendees of
+      Set<String> eventAttendees = event.getAttendees();
+      for (String attendee : attendeesList) {
+        if (eventAttendees.contains(attendee)) {
+          clashAttendeeEvents.add(event.getWhen());
+        }
+      }
+    }
+
+    /* loop through all events attended by this meeting attendees
+    to see if any of the times would clash
+    */
+    
+    // if there are no attendee-clash meetings, then the whole day is available
+    if (clashAttendeeEvents.size() == 0) {
+        if (meetingDuration <= TimeRange.WHOLE_DAY.duration()) {
+          proposedTimeRanges.add(TimeRange.WHOLE_DAY);
+        }
+      return (proposedTimeRanges);
+    }
+
+    // order the clashing events into start time order
+    Collections.sort(clashAttendeeEvents, TimeRange.ORDER_BY_START);
+    TimeRange startTime = TimeRange.fromStartEnd(startDay, clashAttendeeEvents.get(0).start(), false);
+
+    // only add the time if there's enough duration for the requested meeting
+    if (startTime.duration() >= request.getDuration() || Objects.equals(startTime.duration(), request.getDuration())){
```

Comment:
```suggestion
    if (startTime.duration() >= request.getDuration()){
```

As you don't actually care whether the two objects are the same I guess?